### ISSUE TITLE
Aliyun OSS Incompatibility

### ISSUE THREAD

**Issue by GZGavinZhao** [mozilla/sccache#1167]
Hi! I'm using `sccache` with Aliyun Object Storage Service, which is S3-compatible. However, I keep getting cache write errors. I've tested my API credentials and made sure they were working.

I suspect the problem is with the URL. This short issue (kahing/goofys#336) explains it much better than I can. In short, OSS only supports virtual hosted-style URLs, while `sccache` uses path-based access.

**Comment by feuvan** [comment]
OSS is not exactly S3 compatible.

**Comment by GZGavinZhao** [comment]
@feuvan Yes, it's not 100% compatible, but for the `sccache` use case it is enough. #1086 will fix this.

**Comment by Xuanwo** [comment]
I will working on this issue after existing storage services have been migrated to opendal.

I plan to add a new OSS service support via opendal's native support: [opendal::services::oss](https://opendal.databend.rs/opendal/services/oss/index.html)

**Comment by nautaa** [comment]
@Xuanwo Hi, Any progress on this issue？ I'm trying to use OSS now and seeing `SecondLevelDomainForbidden` errors. Then I found this link https://github.com/apache/incubator-opendal/blob/main/core/src/services/s3/compatible_services.md in OpenDAL. Is this a good solution?

**Comment by Xuanwo** [comment]
> @Xuanwo Hi, Any progress on this issue？ I'm trying to use OSS now and seeing `SecondLevelDomainForbidden` errors. Then I found this link https://github.com/apache/incubator-opendal/blob/main/core/src/services/s3/compatible_services.md in OpenDAL. Is this a good solution?

Hi, thanks for the question.

First of all, OSS native support is not added yet. I will continue this work after https://github.com/mozilla/sccache/pull/1885 get merged.

Then, as described in [opendal's docs](https://github.com/apache/incubator-opendal/blob/main/core/src/services/s3/compatible_services.md), we need a new option called `enable_virtual_host_style` to access OSS like `bucket.oss.example.com` instead of `oss.example.com/bucket`. We can address this in https://github.com/mozilla/sccache/issues/1887.

Please let me know if you are interested in this work. I am willing to provide assistance with implementation and review.

**Comment by AJIOB** [comment]
Is it still active? Looks like we’re added the Alibaba OSS support to the sccache
